doc/md/guides/mysql/prefix-indexes.md

```diff
@@ -0,0 +1,451 @@
+```
+
+In the above example we saw how we can improve performance for a `TEXT` type column by specifying appropriate `length` while creating the index.
+
+Now in the next example, let’s see how we can optimize space used by the index with  a prefix index.
+
+### 2. Using prefix indexes to optimize space usage and increase query performance
+
+Using the same table, let’s assume that we want to fetch data of senders by using their email address. Here is a sample query and its cost without any index:
+
+```sql
+SELECT
+    id,
+    sender,
+    sender_address
+FROM
+    fbi_suspects
+WHERE
+    email = 'dsec4@example.com';
+```
+
+```console title="Output"
++--------+----------+----------------------------+
+| id     | sender   | sender_address             |
++--------+----------+----------------------------+
+| 210829 | Jami Lam | 545 North Rocky Milton St. |
++--------+----------+----------------------------+
+```
+
+```console title="Cost"
++-----------------+---------------+
+| Variable_name   | Value         |
++-----------------+---------------+
+| Last_query_cost | 108453.391718 |
++-----------------+---------------+
+```
+
+To optimize this query, usually we would create the following index:
+
+```sql
+CREATE INDEX 
+    email_idx 
+ON 
+    fbi_suspects(email);
+```
+
+```console title="Output"
+Query OK, 0 rows affected (1.57 sec)
+Records: 0  Duplicates: 0  Warnings: 0
+```
+
+The cost for the query after creating this index is as follows:
+
+```console title="Cost"
++-----------------+----------+
+| Variable_name   | Value    |
++-----------------+----------+
+| Last_query_cost | 0.870951 |
++-----------------+----------+
+```
+
+The storage space used by the index is as following:
+
+```sql
+SELECT 
+stat_value
+FROM 
+mysql.innodb_index_stats 
+WHERE 
+index_name = 'email_idx' AND stat_name= 'size';
+```
+
+```console title="Output"
++------------+
+| stat_value |
++------------+
+|       2085 |
++------------+
+```
+
+:::info
+The `innodb_page_size` variable specifies the size of the pages used by the InnoDB storage engine, and the `stat_value` column contains statistics about the index, such as the number of pages used by the index. When `stat_name` is 'size', the `stat_value` column contains the size of the index in terms of number of pages. 
+
+For example, if the `innodb_page_size` system variable is set to 16 KB and the `stat_value` column contains the value 10, this means that the index uses 10 pages, or 160KB of disk space.
+:::
+
+For our example, keeping `innodb_page_size` as 16KB in mind, our index uses ~32MB disk space. Now, we can further improve the query performance as well as reduce the storage used by the index by using a prefix index on the email column. Let’s create a prefix index with the following command:
+
+```sql
+ALTER TABLE fbi_suspects DROP INDEX email_idx;
+CREATE INDEX 
+    email_prefix_idx 
+ON 
+    fbi_suspects(email(5));
```

Comment:
```suggestion
ALTER TABLE fbi_suspects DROP INDEX email_idx;
CREATE INDEX 
    email_prefix_idx 
ON 
    fbi_suspects(email(5));
```

ALTER can have both ADD and DROP. Maybe stay consistent and use one ALTER or DROP and CREATE.